sdk/python/feast/infra/astra.py

```diff
@@ -0,0 +1,146 @@
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+import pandas
+from tqdm import tqdm
+
+from feast import FeatureTable
+from feast.entity import Entity
+from feast.feature_view import FeatureView
+from feast.infra.offline_stores.offline_utils import get_offline_store_from_config
+from feast.infra.online_stores.helpers import get_online_store_from_config
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.infra.provider import (
+    Provider,
+    RetrievalJob,
+    _run_field_mapping,
+    _convert_arrow_to_proto,
+    _get_column_names,
+)
+from feast.registry import Registry
+from feast.repo_config import RepoConfig
+
+
+class AstraProvider(Provider):

```

Comment:
Note: you can use https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/passthrough_provider.py now to reduce this duplication

Reply:
Thanks for sharing this, I will also Extend from Pass-through provider class